SubmittingPatches-backports.rst

```diff
@@ -0,0 +1,344 @@
+in master. In that case there are two possibilities: either the bug was fixed by
+one or more commits in master, or the relevant code in master got refactored,
+but the refactoring itself is not a candidate for backport.
+
+In the former case, just create a master tracker issue citing the PR or commits
+that need to be cherry-picked, populate the Backport field, and proceed as with
+any other backport.
+
+In the latter case (master code was refactored), implement the fix in the most
+recent stable branch that has the bug, and cherry-pick it from there to the
+older stable branches. In each commit, include a brief statement explaining why
+the change cannot be cherry-picked from master. For example::
+
+    This commit cannot be cherry-picked from master because the 
+    code in src/foo/bar.cc was refactored post-nautilus.
+
+
+The ceph-backport.sh script
+---------------------------
+
+Similar to the case of creating the Backport tracker issues, staging the actual
+backport PR and updating the Backport tracker issue is difficult - if not
+impossible - to get right if you're doing it manually, and quickly becomes
+tedious if you do it more than once in a long while.
+
+After having spent a lot of time and effort developing a fix and getting it
+merged into master, it wouldn't be surprising if you wanted to get it backported
+to the stable branches with as little fuss as possible. If that's the case,
+you're in luck! There is a script that automates the entire process of
+cherry-picking the commits from the master PR, opening the GitHub backport PR,
+and cross-linking the GitHub backport PR with the correct Backport tracker
+issue. The script can also be used to good effect if you prefer to prepare the
+backport branch yourself.
+
+The script used to create backport issues is located at
+``src/script/ceph-backport.sh`` in the master branch. Though there might be
+an older version of this script in a stable branch, do not use it. Only use the
+most recent version from master.
+
+This is just a bash script, so the only dependency is ``bash`` itself, but it
+does need to be run in the top level of a local clone of ``ceph/ceph.git``.
+A small up-front time investment is required to get the script working in your
+environment. This is because the script needs to autenticate itself (i.e., as
+you) in order to use the GitHub and Redmine REST API services.
+
+The script is self-documenting. Just run the script and proceed from there.
+
+Once the script has been set up properly, you can validate the setup like so::
+
+    ceph-backport.sh --setup
+
+Once you have this saying "Overall setup is OK", you have two options for
+staging the backport: either leave everything to the script, or prepare the
+backport branch yourself and use the script only for creating the PR and
+updating the Backport tracker issue.
+
+If you prefer to leave everything to the script, just provide the Backport
```

Comment:
```suggestion
If you prefer leaving everything to the script, just provide the Backport
```

Reply:
@tchaikov "prefer" can take a gerund or an infinitive - see https://www.wordsmyth.net/blog/2013/12/prefer/

To a native speaker, there is a subtle difference.